### ISSUE TITLE
Outdated scheduling in "Design Principles"

### ISSUE THREAD

**Issue by tozaicevas** [reactjs/reactjs.org#3474]
> This is a subtle distinction but a powerful one. Since you don’t call that component function but let React call it, it means React has the power to delay calling it if necessary. **In its current implementation React walks the tree recursively and calls render functions of the whole updated tree during a single tick**. However in the future it might start delaying some updates to avoid dropping frames.

> If something is offscreen, we can delay any logic related to it. ... To be clear, we are not taking advantage of this right now.

As far as I understand, React **is taking** advantage of delaying updates? https://github.com/facebook/react/issues/6170#issuecomment-318204454
If that's the case, seems like this part of the docs is outdated.

**Comment by aniledev** [comment]
Yeah, the [legacy docs on scheduling](https://legacy.reactjs.org/docs/design-principles.html#scheduling) aren't being updated anymore. The new docs are [here ](https://react.dev/)as I'm sure you know. =) Might be helpful for anyone who stumble across this issue like me.
